lib/pf/UnifiedApi.pm

```diff
@@ -0,0 +1,183 @@
+    { controller => 'reports', collection_v2a => { get => 'connectiontype_active' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/connectiontype_active' },
+    { controller => 'reports', collection_v2a => { get => 'connectiontypereg_all' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/connectiontypereg_all' },
+    { controller => 'reports', collection_v2a => { get => 'connectiontypereg_active' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/connectiontypereg_active' },
+    { controller => 'reports', collection_v2a => { get => 'ssid' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/ssid' },
+    { controller => 'reports', collection_v2a => { get => 'ssid_all' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/ssid_all' },
+    { controller => 'reports', collection_v2a => { get => 'ssid_active' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/ssid_active' },
+    { controller => 'reports', collection_v2a => { get => 'osclassbandwidth' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/osclassbandwidth' },
+    { controller => 'reports', collection_v2a => { get => 'osclassbandwidth_all' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/osclassbandwidth_all' },
+    { controller => 'reports', collection_v2a => { get => 'nodebandwidth' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/nodebandwidth' },
+    { controller => 'reports', collection_v2a => { get => 'nodebandwidth_all' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/nodebandwidth_all' },
+    { controller => 'reports', collection_v2a => { get => 'topsponsor_all' }, resource_v2a => {},  id_key => 'report_id', path => '/reports/topsponsor_all' },
+
+    { controller => 'nodes', collection_v2a => { get => 'latest_locationlog_by_mac' }, resource_v2a => {}, path => '/nodes/:mac/latest_locationlog' },
+    { controller => 'nodes', collection_v2a => { get => 'locationlog_by_mac' }, resource_v2a => {}, path => '/nodes/:mac/locationlog' },
+);
+
+sub startup {
+    my ($self) = @_;
+    $self->routes->namespaces(['pf::UnifiedApi::Controller', 'pf::UnifiedApi']);
+    $self->hook(before_dispatch => \&set_tenant_id);
+    $self->plugin('pf::UnifiedApi::Plugin::RestCrud');
+    $self->setup_api_v1_routes();
+    $self->custom_startup_hook();
+}
+
+sub setup_api_v1_routes {
+    my ($self) = @_;
+    my $r = $self->routes;
+    my $api_v1_route = $r->any("/api/v1")->name("api.v1");
+    foreach my $route ($self->api_v1_routes) {
+        $api_v1_route->rest_routes($route);
+    }
+
+    my $user_route = $api_v1_route->find('api.v1.Users.resource');
+    $r->any(sub {
+        my ($c) = @_;
+        return $c->render(json => { message => "Unknown path", errors => [] }, status => 404);
+    });
+}
+
+sub api_v1_routes {
+    my ($self) = @_;
+    return @API_V1_ROUTES, $self->api_v1_custom_routes;
+}
+
+sub api_v1_custom_routes {
```

Comment:
@jrouzierinverse 

Do we need this or should this be stripped out from the code?

Reply:
Yes we need this because if we do not have it then we cannot allow custom route/controllers in our API.